src/wasm-exts/customModuleDecode.ml

```diff
@@ -45,6 +51,7 @@ let peek s = if eos s then None else Some (read s)
 let get s = check 1 s; let b = read s in skip 1 s; b
 let get_string n s = let i = pos s in skip n s; String.sub s.bytes i n
 
+let checkpoint s = let p = !(s.pos) in fun () -> s.pos := p
```

Comment:
Nice!
```suggestion
let checkpoint s = let p = !(s.pos) in fun () -> s.pos := p

```

Reply:
Actually that whole module could use a rewrite since it's a bit precious about where custom sections can occur. I opened an issue for that #2890